WebsocketSocket.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * Hoa
+ *
+ *
+ * @license
+ *
+ * New BSD License
+ *
+ * Copyright © 2007-2015, Hoa community. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Hoa nor the names of its contributors may be
+ *       used to endorse or promote products derived from this software without
+ *       specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Websocket;
+
+use Hoa\Core;
+use Hoa\Socket;
+
+/**
+ * Class \Hoa\Websocket\Socket.
+ *
+ * Websocket specific socket extension.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+class WebsocketSocket extends Socket
+{
+    /**
+     * Secured ?
+     *
+     * @var boolean
+     */
+    protected $_secure   = false;
+
+    /**
+     * Endpoint.
+     *
+     * @var string
+     */
+    protected $_endPoint = '/';
+
+    /**
+     * Constructor
+     *
+     * @param string  $uri      Socket URI
+     * @param boolean $secure   Secure mode
+     * @param string  $endPoint Websocket endpoint
+     */
+    public function __construct($uri, $secure = false, $endPoint = '/')
+    {
+        parent::__construct($uri);
```

Comment:
Maybe it can...

I submitted something similar here :

* https://github.com/shulard/Websocket/blob/feature/application-transport-layer/Socket.php#L67-L83
* https://github.com/shulard/Websocket/blob/feature/application-transport-layer/Connection/Connection.php

The second update on `Connection` is just an extension of the `Socket\Connection` to use the specific Socket extension... But we lose the registered wrappers here... Maybe we can make a mix.